### ISSUE TITLE
Calendar: Exception when min or max becomes a non-valid date

### ISSUE THREAD

**Issue by udos86** [telerik/kendo-angular#242]
`CalendarComponent` should be capable of handling `min` and `max` property bindings that are / become `null` instead of throwing an exception.  

This is an essential feature to any [dynamic form approach](https://github.com/udos86/ng2-dynamic-forms).

At the moment the following exception is caused:

```
Error: Expression has changed after it was checked. Previous value: 'true'. Current value: 'false'.
    at ExpressionChangedAfterItHasBeenCheckedError.d (zone.min.js:1)
    at ExpressionChangedAfterItHasBeenCheckedError.BaseError [as constructor] (core.umd.js:1186)
    at new ExpressionChangedAfterItHasBeenCheckedError (core.umd.js:6607)
    at checkBinding (core.umd.js:6784)
    at Wrapper_NgControlStatusGroup.checkHost (/InternalFormsSharedModule/NgControlStatusGroup/wrapper.ngfactory.js:43)
    at CompiledTemplate.proxyViewClass.View_n0.detectChangesInternal (/e/n/component.ngfactory.js:1844)
    at CompiledTemplate.proxyViewClass.AppView.detectChanges (core.umd.js:12738)
    at CompiledTemplate.proxyViewClass.DebugAppView.detectChanges (core.umd.js:12885)
    at CompiledTemplate.proxyViewClass.AppView.internalDetectChanges (core.umd.js:12723)
    at View_KendoExampleComponent1.detectChangesInternal (/AppModule/KendoExampleComponent/component.ngfactory.js:166)
    at View_KendoExampleComponent1.AppView.detectChanges (core.umd.js:12738)
    at View_KendoExampleComponent1.DebugAppView.detectChanges (core.umd.js:12885)
    at ViewContainer.detectChangesInNestedViews (core.umd.js:12996)
    at CompiledTemplate.proxyViewClass.View_KendoExampleComponent0.detectChangesInternal (/AppModule/KendoExampleComponent/component.ngfactory.js:300)
    at CompiledTemplate.proxyViewClass.AppView.detectChanges (core.umd.js:12738) 
```

**Comment by ggkrustev** [comment]
Hi @udos86,
Indeed, the component will throw an exception when min/max are nulls (we need a better exception handling here). Generally speaking, those values are *required*, because the component needs to know the range in which it will operate.

Before we discuss possible workarounds/fixes, could you let me know whether someone will need to set the min/max range to `null`? Does the `null` mean reset to default component state or remove the range limit, introducing an infinite calendar?

**Comment by udos86** [comment]
@ggkrustev First of all, let me say, that I really like the Kendo UI components for Angular 2! They're a high quality work only comparable to Angular 2 Material at the moment. 

That's why I'd like to deliver first class support in my widely used [dynamic forms library](https://github.com/udos86/ng2-dynamic-forms) mentioned above.

> Before we discuss possible workarounds/fixes, could you let me know whether someone will need to set the min/max range to null? Does the null mean reset to default component state or remove the range limit, introducing an infinite calendar?

I'd say the component should introduce an infinite calendar then. That would be the expected behaviour.

The thing here is, that `min` and `max` bindings are optional. But when working with a [dynamic forms approach](https://angular.io/docs/ts/latest/cookbook/dynamic-form.html) the binding needs to be set in the component template regardless if the user assigns min or max properties in it's model definition.

**Comment by ggkrustev** [comment]
@udos86, thanks for the nice words and for the clarification.

As I mentioned, throwing an exception is not the best solution, although a predefined min/max range is required. That being said, [a short research on the subject](https://gist.github.com/ggkrustev/fe741b95e79ac3e83eec024420fd673b) showed that we just need to use the default range if `min` or `max` are set to *null*. In other words:
- when `min` == null -> use the default Calendar min value
- when `max` == null -> use the default Calendar max value

The changes will be available in the next release of the package. Once the changes are live, I will close the issue.

[TL;DR]
In general, introducing an infinite calendar scrolling looks tempting at first, as it is really cool feature. The implementation hurdles, however, are not easy to overcome, not to mention the performance hit that will come with the numerous virtual list recreation. To be honest, I cannot justify such exponential complexity increase in the implementation just to support infinite calendar date range. That being said, a fallback to the default values is much better solution.

**Comment by udos86** [comment]
@ggkrustev That's great! Thanks for the enhancement!

**Comment by ggkrustev** [comment]
No problem at all! Thank you for the feedback in first place :). The feature will released till the end of the work week (it's in review stage).

**Comment by ggkrustev** [comment]
Available in `@progress/kendo-angular-dateinputs@0.3.1`.
